### PR TITLE
Reduce exif provider errors during file scan

### PR DESCRIPTION
**Pull request by artonge**
Update metadata logic a bit to limits errors during file scan.

1. Prevent inserting when a metadata already exists to not create error in the middle of a transaction. This causes issue on Postgres.
2. Reduce EXIF data extraction to a subset of image file formats. This will reduce the amount of `exif_read_data(): File not supported` error.

### CODE REVIEW

**Review by blizzz — CHANGES_REQUESTED** [review]
one question

**Review by blizzz — CHANGES_REQUESTED** [review]
one question

**Comment by blizzz** [comment]
(double review powered by :window:  :octocat:  :unicorn: )

File: lib/private/Metadata/FileMetadataMapper.php [hunk 1]
```diff
@@ -148,4 +148,30 @@ public function update(Entity $entity): FileMetadata {
...
+	 * Override the insertOrUpdate as we could be in a transaction in which case we can not afford on error.
+	 *
+	 * @param FileMetadata $entity the entity that should be created/updated
+	 * @return FileMetadata the saved entity with the (new) id
+	 * @throws Exception
+	 * @throws \InvalidArgumentException if entity has no id
+	 */
+	public function insertOrUpdate(Entity $entity): FileMetadata {
+		try {
+			$existingEntity = $this->findForGroupForFile($entity->getId(), $entity->getGroupName());
```
[the remark this answers is not in the document]
Comment (artonge) [reply lib/private/Metadata/FileMetadataMapper.php:162]: True, I replaced the `$this->insert` call to `parent::insertOrUpdate` to take advantage of the error handling for DBs that supports it.

**Review by blizzz — APPROVED** [review]

**Review by come-nc — APPROVED** [review]

**Pull request merged**